packages/as-proto-gen/src/index.ts

```diff
@@ -1,60 +1,95 @@
+import * as assert from "assert";
+import * as fs from "fs";
 import {
   CodeGeneratorRequest,
   CodeGeneratorResponse,
 } from "google-protobuf/google/protobuf/compiler/plugin_pb";
+
+import { FileContext } from "./file-context";
+import { processFile, addFile, generateExport } from "./generate/file";
 import { GeneratorContext } from "./generator-context";
-import { generateFile } from "./generate/file";
 import { getPathWithoutProto } from "./names";
-import { FileContext } from "./file-context";
-import prettier from "prettier";
-import * as fs from "fs";
-import * as assert from "assert";
 
-const input = fs.readFileSync(process.stdin.fd);
+fs.readFile(process.stdin.fd, (err, input) => {
+  if (err !== null) {
+    console.log("An error occurred in as-proto generator plugin.");
+    console.error(err);
+    process.exit(1);
+  }
+
+  try {
+    const codeGenRequest = CodeGeneratorRequest.deserializeBinary(input);
+    const codeGenResponse = new CodeGeneratorResponse();
+    const generatorContext = new GeneratorContext();
 
-try {
-  const codeGenRequest = CodeGeneratorRequest.deserializeBinary(input);
-  const codeGenResponse = new CodeGeneratorResponse();
-  const generatorContext = new GeneratorContext();
+    const PROTOC_VERSION = codeGenRequest
+      .getCompilerVersion()
+      ?.toArray()
+      .slice(0, 3)
```

Comment:
why slice?

Reply:
not including the suffix return by getCompilerVersion. Most of the time this will be empty and didn't wanted version ending with a dot.  ie: `3.20.1.`